garden-service/src/commands/deploy.ts

```
@@ -69,10 +72,23 @@ export class DeployCommand extends Command<Args, Opts> {
     const serviceNames = getNames(services)
 
     if (services.length === 0) {
-      garden.log.warn({ msg: "No services found. Aborting." })
+      garden.log.error({ msg: "No services found. Aborting." })
       return { result: {} }
     }
 
+    let watch
+    const hotReloadServiceNames = opts["hot-reload"] || []
+    const hotReloadModuleNames = await getHotReloadModuleNames(garden, hotReloadServiceNames)
+
+    if (opts["hot-reload"]) {
+      if (!validateHotReloadOpt(garden, hotReloadServiceNames)) {
```

Should this throw an error instead of just returning?

---

Yeah, good point. Will make that change.